src/Spectre.libWavelet/WaveletUtils.h

```diff
@@ -0,0 +1,60 @@
+/*
+* WaveletUtils.h
+* Common functions used by wavelet decomposition and reconstruction.
+*
+Copyright 2018 Michal Gallus
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+#include "AlgorithmConstants.h"
+
+namespace spectre::algorithm::wavelet
+{
+/// <summary>
+/// Computes the current scale of the wavelet.
+/// </summary>
+/// <param name="level">Wavelet level.</param>
+/// <returns>Scale.</returns>
+inline size_t ComputeScale(unsigned level)
+{
+    return 1ULL << (level + 1); // 2^(L + 1)
+}
+
+/// <summary>
+/// Performs a ceiled divison on integers.
+/// </summary>
+/// <param name="dividend">The nominator.</param>
+/// <param name="divisor">The denominator.</param>
+/// <returns>Result of ceil(dividend/divisor).</returns>
+inline size_t CeiledDivision(size_t dividend, size_t divisor)
+{
+    return dividend / divisor + (dividend % divisor != 0);
```

Comment:
You're not checking the case when `divisor` is 0.

Reply:
That's right. I wanted to avoid unnecessary checks to gain performance.

Reply:
I've added a comment to summary that it doesn't check against divisor = 0, so that if user feels that's necessary for him, he can perform the check himself.

Reply:
> if user feels that's necessary for him

Exceptions are for the programmer.

Reply:
Think I'll just inline that function by hand then.

Reply:
It has been agreed that the function will gain postfix "Unsafe" to indicate that it doesn't perform any argument checks.